pkg/getter/httpgetter.go

```diff
@@ -90,7 +90,9 @@ func NewHTTPGetter(options ...Option) (Getter, error) {
 }
 
 func (g *HTTPGetter) httpClient() (*http.Client, error) {
-	if (g.opts.certFile != "" && g.opts.keyFile != "") || g.opts.caFile != "" {
+	client := http.DefaultClient
+	switch {
+	case (g.opts.certFile != "" && g.opts.keyFile != "") || g.opts.caFile != "":
```

Comment:
I am not entirely sure this was correct in the old version or the new version. I was under the impression that InsecureSkipVerify did not mean that the certificates should be completely omitted -- just that they would not be verified against the CA file (or against the host-oriented fields). But are they completely unused altogether?

This shouldn't block this PR, since this PR is a more idiomatic version of the old code... but if anyone knows more than me about this, it would be good to get an opinion over whether certs vs InsecureSkipVerify is correct.

Reply:
This was also pointed out in #8031 if you want to have a look there :)